library/core/src/fmt/mod.rs

```diff
@@ -2233,35 +2233,40 @@ impl Display for char {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T: ?Sized> Pointer for *const T {
     fn fmt(&self, f: &mut Formatter<'_>) -> Result {
-        /// Since the formatting will be identical for all pointer types, use a non-monomorphized
-        /// implementation for the actual formatting to reduce the amount of codegen work needed
-        fn inner(ptr: *const (), f: &mut Formatter<'_>) -> Result {
-            let old_width = f.width;
-            let old_flags = f.flags;
-
-            // The alternate flag is already treated by LowerHex as being special-
-            // it denotes whether to prefix with 0x. We use it to work out whether
-            // or not to zero extend, and then unconditionally set it to get the
-            // prefix.
-            if f.alternate() {
-                f.flags |= 1 << (FlagV1::SignAwareZeroPad as u32);
-
-                if f.width.is_none() {
-                    f.width = Some((usize::BITS / 4) as usize + 2);
-                }
-            }
-            f.flags |= 1 << (FlagV1::Alternate as u32);
+        pointer_fmt_inner((*self as *const ()).addr(), f)
```

Comment:
I'm not sure if this `as *const ()` does anything. 

@jam1garner do you have an idea how this might affect anything/how to test this?

Reply:
Can you say more about why it's here? I would expect this code to be self.addr(), I think?

Reply:
I've left it here from the previous version in which it was used to erase the type. But since this gets the addr, I guess it's useless now.

Reply:
The reason it is there is in order to erase the type to avoid monomorphizing the impl, since the debug/pointer formatting doesn't care about the type, just the address.

The reason this matters is that if you derive(Debug) for bindgen'd bindings involving lots of pointer fields it generates an obscene amount of code, and severely hurts compile times.

I agree that it should be entirely useless now in the presence of .addr()!

Reply:
I think the `T: Sized` bound on `addr` exists to make discarding the metadata explicit.

Reply:
`addr()` already discards other information so I don't disagree -- however, `map_addr` should then probably also be made to support `?Sized`.

@Gankra designed those APIs so she might better be able to explain why `addr` requires `Sized`.